modules/postprocessing/include/modules/postprocessing/processors/depthoffield.h

```diff
@@ -0,0 +1,150 @@
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#pragma once
+
+#include <modules/postprocessing/postprocessingmoduledefine.h>
+#include <inviwo/core/common/inviwo.h>
+#include <inviwo/core/datastructures/camera/perspectivecamera.h>
+#include <inviwo/core/datastructures/camera/skewedperspectivecamera.h>
+#include <inviwo/core/datastructures/image/imageram.h>
+#include <inviwo/core/datastructures/image/layerramprecision.h>
+#include <inviwo/core/datastructures/volume/volumeram.h>
+#include <inviwo/core/datastructures/volume/volumeramprecision.h>
+#include <inviwo/core/ports/imageport.h>
+#include <inviwo/core/ports/meshport.h>
+#include <inviwo/core/processors/processor.h>
+#include <inviwo/core/properties/cameraproperty.h>
+#include <inviwo/core/properties/eventproperty.h>
+#include <inviwo/core/properties/ordinalproperty.h>
+#include <modules/base/algorithm/randomutils.h>
+#include <modules/opengl/image/imagegl.h>
+#include <modules/opengl/shader/shader.h>
+#include <modules/opengl/shader/shaderutils.h>
+#include <modules/opengl/texture/textureutils.h>
+#include <modules/opengl/volume/volumeutils.h>
+#include <modules/opengl/volume/volumegl.h>
+
+namespace inviwo {
+
+/** \docpage{org.inviwo.DepthOfField, Depth Of Field}
+ * ![](org.inviwo.DepthOfField.png?classIdentifier=org.inviwo.DepthOfField)
+ *
+ * Use any pinhole image as input and connect the camera property to the rendering processor.
+ * Applies a depth of field effect where out-of-focus regions appear blurred. The image should be
+ * generated with a perspective camera but may use any rendering technique.
+ *
+ * ### Inports
+ *   * __ImageInport__ Input image.
+ *   * __MeshInport__ World space point to focus on. The first point in the first position buffer
```

Comment:
This is the current solution suggested by Tino (my supervisor in this project). We didn't want to use vec3 since we couldn't find interesting processors with vec3 outports.

Reply:
Ok, even though it is a bit more work on the network level it might be better to create a processor that extracts a point from a mesh (or buffer or data frame)

Reply:
Okay, once that is done, it will be very straightforward to modify this processor. I'm not sure what kind of data might be interesting so I think it's out of scope for this PR.